src/raft.rs

```diff
@@ -1260,9 +1248,8 @@ impl<T: Storage> Raft<T> {
         };
 
         self.set_pending_membership_change(conf_change.clone());
-        let max_inflights = self.max_inflight;
-        self.mut_prs()
-            .begin_membership_change(configuration, Progress::new(1, max_inflights))?;
+        let pr = Progress::new(self.raft_log.last_index() + 1, self.max_inflight);
```

Comment:
Keep same with `add_voter` and `and_learner`. BTW some test cases will fail because on leader entry 1 is considered as "compacted" so that followers need to retrieve snapshots. But in test code we don't handle snapshots.